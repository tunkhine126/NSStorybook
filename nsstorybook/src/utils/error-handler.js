/* eslint-disable no-unused-vars */
import React from 'react';
import NotFound from 'components/NotFound';

// Handle Custom network errors
export const networkErrors = ({
  graphQLErrors,
  networkError,
  operation,
  response,
}) => {
  if (graphQLErrors) {
    graphQLErrors.map(({ message, locations, path }) => ({
      type: 'GraphQL error',
      message,
      locations,
      path,
    }));
  }

  if (networkError) {
    // console.log(`[Network error]: ${networkError}`);
  }
};

// Handles All UI Network Errors
export const generalErrorHandler = (error, { ...rest }) => {
  // Handle Unauthorized
  if (error.message.includes('Unauthorized')) {
    return <NotFound unAuthorized {...rest} />;
  }
  // Handle Failed Queries
  return <NotFound failedQuery {...rest} message={error.message} />;
};
