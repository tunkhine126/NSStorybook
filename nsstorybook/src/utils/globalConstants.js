import React from 'react';
import { FormattedMessage } from 'react-intl';
import messages from '../messages';

export const RESTART_ON_REMOUNT = '@@saga-injector/restart-on-remount';
export const DAEMON = '@@saga-injector/daemon';
export const ONCE_TILL_UNMOUNT = '@@saga-injector/once-till-unmount';
export const BREAD_CRUMB_NAME_MAP = {
  '/dashboard': <FormattedMessage {...messages.rtDashboard} />,
  'https://intercom.help/new-story-charity/en/': (
    <FormattedMessage {...messages.needHelp} />
  ),
  '/surveys': <FormattedMessage {...messages.rtSurveys} />,
  '/surveys/templates': <FormattedMessage {...messages.rtTemplates} />,
  '/surveys/templates/survey': <FormattedMessage {...messages.survey} />,
  '/surveys/survey-templates': (
    <FormattedMessage {...messages.rtSurveyTemplates} />
  ),
  '/surveys/submissions': <FormattedMessage {...messages.rtSubmissions} />,
  // '/surveys/requests': <FormattedMessage {...messages.rtRequests} />,
  '/reports': <FormattedMessage {...messages.rtReports} />,
  '/recipients': <FormattedMessage {...messages.rtRecipients} />,
  '/recipients/add/recipient': 'Add Recipient',
  '/settings': <FormattedMessage {...messages.rtSettings} />,
  '/org': <FormattedMessage {...messages.rtOrg} />,
  '/account': <FormattedMessage {...messages.rtAccount} />,
  '/account/account-settings': <FormattedMessage {...messages.rtAccSettings} />,
  '/account/password-settings': (
    <FormattedMessage {...messages.rtPwdSettings} />
  ),
  '/login': <FormattedMessage {...messages.rtLogout} />,
};

// TODO: Enable Disabled Routes

export const ROUTES = {
  account: {
    path: '/account',
    children: [
      {
        name: 'account-settings',
        path: '/account/account-settings',
      },
      {
        name: 'password-settings',
        path: '/account/password-settings',
      },
    ],
  },
  dashboard: {
    path: '/dashboard',
  },
  help: {
    path: 'https://intercom.help/new-story-charity/en/',
    options: {
      es: 'https://intercom.help/new-story-charity/es/',
    },
    active: true,
  },
  login: {
    path: '/login',
  },
  forgotPassword: {
    path: '/forgot-password',
  },
  surveys: {
    path: '/surveys',
    children: [
      {
        name: 'templates',
        path: '/surveys/templates',
      },
      {
        name: 'survey-templates',
        path: '/surveys/survey-templates/:id',
        parentActive: true,
        hideRoute: true,
      },
      {
        name: 'submissions',
        path: '/surveys/submissions',
      },
      // {
      //   name: 'requests',
      //   path: '/surveys/requests',
      // },
    ],
  },
  recipients: {
    path: '/recipients',
    children: [
      {
        name: 'add-recipient',
        path: '/recipients/add/recipient',
        hideRoute: true,
      },
    ],
  },
  reports: {
    path: '/reports',
  },
  settings: {
    path: '/settings',
  },
  org: {
    path: '/org',
  },
};

export const PARENT_ROUTES = {
  top: [
    // { display: 'dashboard' },
    { display: 'surveys', hasChildren: true },
    { display: 'recipients', hasChildren: true },
    // { display: 'reports' },
    // { display: 'settings' },
  ],
  bottom: [
    { display: 'org' },
    { display: 'account', hasChildren: true },
    { display: 'help' },
    { display: 'login' },
  ],
};

export const LOCAL_STORAGE = {
  userToken: 'NS_User_Token',
  lastAppVersionUsed: 'NS_Last_App_Version_Used',
  user: 'NS_User',
  org: 'NS_User_Org',
  language: 'NS_User_Lang',
  recipientDefinitionsLoaded: 'NS_RD_Loaded',
};

export const DATA_TYPES = {
  shortText: 'short_text',
  paragraphText: 'paragraph_text',
  number: 'number',
  date: 'date',
  time: 'time',
  singleSelect: 'single_select',
  multiSelect: 'multi_select',
  likert: 'likert',
  photo: 'photo',
  audio: 'audio',
  video: 'video',
  coordinates: 'coordinates',
};

export default {
  DEV_URI: 'https://api-dev.newstory.io/graphql',
  PROD_URI: 'https://api.newstory.io/graphql',
  DEV_X_Api_Key: '0123456789abcdef0123456789abcdef',
  PROD_X_Api_Key: 'f45be18e6cd739d516ef406cbcbc3262',
};
