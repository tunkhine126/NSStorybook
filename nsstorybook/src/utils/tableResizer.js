/* eslint-disable no-unused-vars */
/* eslint-disable no-shadow */
/* eslint-disable no-plusplus */
/* eslint-disable no-void */
/* eslint-disable no-return-assign */
/* eslint-disable prettier/prettier */
/* eslint-disable no-unused-expressions */
/* eslint-disable no-sequences */

export default function resizableGrid(e, color) {
  const t = e.getElementsByTagName('tr')[0];
  const n = t ? t.children : void 0;
  const eventsToRemove = [];
  if (n) {
    for (let i = e.offsetHeight, o = 0; o < n.length - 1; o++) {
      const r = s(i);
      r.style.borderRight = `1px solid ${color}`;
      r.style.marginRight = '5px';
      n[o].appendChild(r), eventsToRemove.push(d(r));
    }
  }
  function d(e) {
    let t;
    let n;
    let i;
    let o;
    let r;

    const mouseDownEvent = e => {
      (n = e.target.parentElement), (i = n.nextElementSibling), (t = e.pageX);
      const d = (function(e) {
        if (l(e, 'box-sizing') === 'border-box') return 0;
        const t = l(e, 'padding-left');
        const n = l(e, 'padding-right');
        return parseInt(t) + parseInt(n);
      })(n);
      (o = n.offsetWidth - d), i && (r = i.offsetWidth - d);
    };

    const mouseMoveEvent = e => {
      if (n) {
        const d = e.pageX - t;
        i && (i.style.width = `${r - d}px`), (n.style.width = `${o + d}px`);
      }
    };

    const mouseUpEvent = e => {
      (n = void 0), (i = void 0), (t = void 0), (r = void 0), (o = void 0);
    };

    e.addEventListener('mousedown', mouseDownEvent),
    document.addEventListener('mousemove', mouseMoveEvent),
    document.addEventListener('mouseup', mouseUpEvent);

    // Return events to remove on component dismount (preventing memory leaks)
    return [
      {
        event: 'mousedown',
        element: e,
        fn: mouseDownEvent,
      },
      {
        event: 'mousemove',
        element: document,
        fn: mouseMoveEvent,
      },
      {
        event: 'mouseup',
        element: document,
        fn: mouseUpEvent,
      },
    ];
  }
  function s(e) {
    const t = document.createElement('div');
    return (
      (t.style.top = 0),
      (t.style.right = 0),
      (t.style.width = '5px'),
      (t.style.position = 'absolute'),
      (t.style.cursor = 'col-resize'),
      (t.style.userSelect = 'none'),
      (t.style.height = `100%`),
      t
    );
  }
  function l(e, t) {
    return window.getComputedStyle(e, null).getPropertyValue(t);
  }

  const dismount = () => {
    eventsToRemove.forEach(column => {
      column.forEach(({ element, event, fn }) => {
        element.removeEventListener(event, fn);
      });
    });
  };

  return {
    dismount,
  };
}
