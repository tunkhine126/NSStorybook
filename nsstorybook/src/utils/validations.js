import React from 'react';
import { FormattedMessage } from 'react-intl';
import messages from '../messages';

export const required = value => {
  if (!value) {
    return <FormattedMessage {...messages.requiredField} />;
  }
  return undefined;
};
