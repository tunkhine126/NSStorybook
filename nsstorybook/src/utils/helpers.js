import Cookies from 'js-cookie';
import { LOCAL_STORAGE } from '../utils/globalConstants';
import moment from 'moment';

export const setLocalStorage = (key, value) => localStorage.setItem(key, value);

export const getLocalStorage = key => localStorage.getItem(key);

export const userTokenExist = () =>
  Cookies.get(LOCAL_STORAGE.userToken) !== undefined;

export const resetNestedLevels = options => {
  const keys = Object.keys(options).filter(
    k => k === 'checked' || k === 'active' || k === 'toggled'
  );

  const resetKeys = obj =>
    keys.forEach(key => {
      obj[key] = false;
    });

  const recursive = function(arr) {
    arr.forEach(obj => {
      resetKeys(obj);
      if (obj.children.length > 0) {
        recursive(obj.children);
      }
    });
  };

  if (options.children.length > 0) {
    recursive(options.children);
  }

  resetKeys(options);
};

export const generateISODate = days =>
  moment(moment(new Date()).format('YYYY-MM-DD'))
    .subtract(days, 'days')
    .toISOString();

export const generateImg = img => {
  const dataSignature = img.includes('data:image/')
    ? ''
    : 'data:image/png;base64,';
  return img.includes('http') ? img : `${dataSignature}${img}`;
};

export const filterObj = (obj, predicate) =>
  Object.keys(obj)
    .filter(key => predicate(key))
    .reduce((res, key) => Object.assign(res, { [key]: obj[key] }), {});
