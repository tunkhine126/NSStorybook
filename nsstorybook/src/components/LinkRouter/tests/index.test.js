/**
 *
 * Tests for LinkRouter
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import { ROUTES } from 'utils/globalConstants';

import LinkRouter from '../index';

describe('<LinkRouter />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <MemoryRouter>
        <LinkRouter color="inherit" to={ROUTES.dashboard.path}>
          Dashboard
        </LinkRouter>
      </MemoryRouter>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <MemoryRouter>
        <LinkRouter color="inherit" to={ROUTES.dashboard.path}>
          Dashboard
        </LinkRouter>
      </MemoryRouter>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
