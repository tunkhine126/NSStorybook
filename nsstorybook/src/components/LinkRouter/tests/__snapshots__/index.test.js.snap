// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkRouter /> Should render and match the snapshot 1`] = `
<a
  class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
  href="/dashboard"
>
  Dashboard
</a>
`;
